src/cache.js

```diff
@@ -0,0 +1,131 @@
+import window from 'global/window';
+import {getUrl} from './url';
+
+// Tracks previously-downloaded scripts and/or detected players.
+//
+// The keys follow the format "accountId_playerId_embedId" where accountId is
+// optional and defaults to "*". This happens when we detect pre-existing
+// player globals.
+const cache = new window.Map();
```

Comment:
maybe this should be called `playerUrlCache` or something more descriptive instead?